tokstr.h

```diff
@@ -2,30 +2,62 @@
 #define __TOKSTR_H
 
 // tokstr -- textual token iterator with some input independence
+// 2022-01-29 [revised during code review, add regions]
 // 2022-01-25 [initially released inside dnsdbq]
 
-/* example:
+/* example using heap-allocated strings:
 
 	tokstr_t ts = tokstr_string("this:is+-test");
 	for (char *t; (t = tokstr_next(ts, "-:+")) != NULL; free(t))
 		printf("\t\"%s\"\n", t);
 	tokstr_last(&ts);
 
- * will output "this", "is", and "test".
+ * will output "this", "is", and "test". so will this:
+
+	tokstr_t ts = tokstr_string("this:is+-test");
+	for (char t[100]; tokstr_next_copy(ts, "-:+", t, sizeof t) > 0;)
+		printf("\t\"%s\"\n", t);
+	tokstr_last(&ts);
+
+ * as will this:
+
+	tokstr_t ts = tokstr_string("this:is+-test");
+	for (;;) {
+		struct tokstr_reg t = tokstr_next_region(ts, "-:+");
+		if (t.base == NULL)
+			break;
+		printf("\t\"%*s\"\n", t.size, t.base);
+	}
+	tokstr_last(&ts);
+
  */
 
 // opaque type for iterator state -- never used
 struct tokstr;
 
-// tokstr_buffer -- create an iterator for a counted string
-struct tokstr *tokstr_buffer(const char *, size_t);
+struct tokstr_reg {
+	const char		*base;
+	size_t			size;
+};
+
+// tokstr_region -- create an iterator for a counted string
+struct tokstr *tokstr_region(struct tokstr_reg);
 
 // tokstr_string -- create an iterator for a nul-terminated string
 struct tokstr *tokstr_string(const char *);
 
-// tokstr_next -- return next token from an iterator (caller must free() this)
+// tokstr_string -- create an iterator for a nul-terminated string
+struct tokstr *tokstr_string(const char *);
+
+// tokstr_next -- return next token from an iterator (strndup)
```

Comment:
it's unclear to a user what strndup means here.

Reply:
fixed (6a2069d..5ac38d6)